packages/lib/src/icon_component.rs

```diff
@@ -1,64 +1,94 @@
+use core::str;
+
 use dioxus::prelude::*;
 
 /// Icon shape trait
 pub trait IconShape {
     fn view_box(&self) -> &str;
+    fn width(&self) -> &str;
+    fn height(&self) -> &str;
     fn xmlns(&self) -> &str;
+    fn title(&self) -> &str;
     fn child_elements(&self) -> Element;
-    fn fill_and_stroke<'a>(&self, user_color: &'a str) -> (&'a str, &'a str, &'a str) {
-        ("none", user_color, "0")
-    }
-    fn stroke_linecap(&self) -> &str {
-        "butt"
-    }
-    fn stroke_linejoin(&self) -> &str {
-        "miter"
-    }
+    fn fill(&self) -> &str;
+    fn stroke(&self) -> &str;
+    fn stroke_width(&self) -> &str;
+    fn stroke_linecap(&self) -> &str;
+    fn stroke_linejoin(&self) -> &str;
 }
 
 /// Icon component Props
 #[derive(PartialEq, Props, Clone)]
 pub struct IconProps<T: IconShape + Clone + PartialEq + 'static> {
     /// The icon shape to use.
     pub icon: T,
-    /// The height of the `<svg>` element. Defaults to 20.
-    #[props(default = 20)]
+    /// The id of the `<svg>` element.
+    #[props(default = None)]
+    pub id: Option<String>,
+    /// The height of the `<svg>` element. Defaults to the icon's default height.
+    #[props(default = 0)]
     pub height: u32,
-    /// The width of the `<svg>` element. Defaults to 20.
-    #[props(default = 20)]
+    /// The width of the `<svg>` element. Defaults to the icon's default width.
+    #[props(default = 0)]
     pub width: u32,
```

Comment:
```suggestion
    pub width: Option<u32>,
```

Reply:
Would it be worth changing width and height input to String to allow for unit identifiers? Or even implement it as an Enum?

Reply:
`String` sounds better IMO